accord-core/src/main/java/accord/local/CommandStores.java

```diff
@@ -625,6 +625,10 @@ public AsyncChain<Void> forEach(Consumer<SafeCommandStore> forEach)
         Snapshot snapshot = current;
         for (ShardHolder shard : snapshot.shards)
             list.add(shard.store.build(empty(), forEach));
+
+        if (list.isEmpty())
```

Comment:
I wonder if we should remove the variants of `AsyncChains.reduce` that don't accept `identity` and use this version here. It's a bit brittle, the lack of properly handling no inputs.